### PR TITLE
Docs for Custom RBAC

### PR DESCRIPTION
**Pull request by steplercamp**
## Summary & Motivation

Building docs for custom RBAC

## How I Tested These Changes
Built Locally

### CODE REVIEW

File: docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx [hunk 1]
```diff
@@ -14,32 +14,56 @@ In this guide, we'll cover how RBAC works in Dagster+, how to assign roles to us
...
 
 ### Teams
 
 Dagster+ Pro users can create teams of users and assign default permission sets. Refer to the [Managing teams in Dagster+](/dagster-plus/account/managing-users/managing-teams) guide for more info.
 
+--- 
+## Creating and editing custom roles
+
+In order to create or edit custom roles:
+1. Navigate to the **Organization Settings** page.
```
Comment (gibsondan) [inline docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx:39]: this has a period, the rest do not

File: docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx [hunk 2]
```diff
@@ -14,32 +14,56 @@ In this guide, we'll cover how RBAC works in Dagster+, how to assign roles to us
...
+2. Click the Roles tab
+
+Then to create a role:
+1. Click the **Create new role** button 
+2. Select a name and icon
+3. Provide a brief description
+4. Choose if deployment or organization type depending on where you want your role applied
```
Comment (gibsondan) [inline docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx:46]: stray "if"

File: docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx [hunk 3]
```diff
@@ -14,32 +14,56 @@ In this guide, we'll cover how RBAC works in Dagster+, how to assign roles to us
...
+5. Select the permissions you want to apply to the role. Note - you can base your role off an existing role for ease of creation
```
Comment (gibsondan) [inline docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx:47]: off of

File: docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx [hunk 4]
```diff
@@ -14,32 +14,56 @@ In this guide, we'll cover how RBAC works in Dagster+, how to assign roles to us
...
+2. Make your changes
+3. Save your changes
+4. You will see a confirmation dialog including the changes that have been made
 ---
 
-## Assigning user and team roles
+## Deleting custom roles
+Deleting custom roles is similar to editing. At the bottom of the edit dialog you'll see a button to "Delete role". Two things to note when deleting a role:
+1. Deleting a role is a permanent action and cannot be undone
+2. If the role is currently assigned to any users, you will need to reassign them to a different role before you can delete the role
```
Comment (gibsondan) [inline docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx:59]: users or teams

File: docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx [hunk 5]
```diff
@@ -14,32 +14,56 @@ In this guide, we'll cover how RBAC works in Dagster+, how to assign roles to us
...
-
-For example, the **Admin** user role includes permissions specific to this role and all permissions in the **Editor**, **Launcher**, and **Viewer** user roles. Refer to the [User permissions reference](#user-permissions-reference) for the full list of user permissions in Dagster+.
+1. **Permission Scope** - Permissions are context-specific, falling into two main categories:
+   - Organization-wide settings (for example, "Create Teams")
+   - Deployment-specific actions (for example, "Launch and Cancel Backfills")
+2. **Role Types** - Dagster+ supports two types of roles:
+   - **Default Roles:** Hierarchical roles, based on sensible defaults that come out of the box with Dagster.
+   - **Custom Roles:** Allows you to define tailored roles with specific sets of permissions
 
+This dual approach provides both the simplicity of predefined roles and the flexibility to create custom roles that precisely match your organization's needs.
```
[the remark this answers is not in the document]
Comment (steplercamp) [reply docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx:26]: Sort of both I think (@PedramNavid wrote this part). 

permission scope is an outcome of role types, so i think its hard to disentangle

File: docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx [hunk 6]
```diff
@@ -14,32 +14,56 @@ In this guide, we'll cover how RBAC works in Dagster+, how to assign roles to us
...
 ### User role enforcement
 
-All user roles are enforced both in Dagster+ and the [GraphQL API](/concepts/webserver/graphql).
+All roles are enforced both in Dagster+ and the [GraphQL API](/concepts/webserver/graphql).
```
[the remark this answers is not in the document]
Comment (steplercamp) [reply docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx:29]: Yes, I know that! But figure it's not built yet, and we don't need to advertise that catalog viewer will be slightly different

File: docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx (hunk 1 above)
Comment (steplercamp) [reply docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx:39]: great catch

File: docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx (hunk 5 above)
[the remark this answers is not in the document]
Comment (PedramNavid) [reply docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx:26]: i meant the permission scope and role types but i could be wrong

File: docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx (hunk 4 above)
Comment (steplercamp) [reply docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx:59]: fixed

File: docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx [hunk 7]
```diff
@@ -14,32 +14,56 @@ In this guide, we'll cover how RBAC works in Dagster+, how to assign roles to us
...
-- Launcher (Pro plans only)
-- Viewer
-
-For example, the **Admin** user role includes permissions specific to this role and all permissions in the **Editor**, **Launcher**, and **Viewer** user roles. Refer to the [User permissions reference](#user-permissions-reference) for the full list of user permissions in Dagster+.
+1. **Permission Scope** - Permissions are context-specific, falling into two main categories:
+   - Organization-wide settings (for example, "Create Teams")
+   - Deployment-specific actions (for example, "Launch and Cancel Backfills")
+2. **Role Types** - Dagster+ supports two types of roles:
+   - **Default Roles:** Hierarchical roles, based on sensible defaults that come out of the box with Dagster.
+   - **Custom Roles:** Allows you to define tailored roles with specific sets of permissions
```
[the remark this answers is not in the document]
Comment (steplercamp) [reply docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx:24]: @PedramNavid , do you have any guidance here. Do we have a word for self-serve?

I think I could keep the previous doc under a Dagster+ self-serve, then add a section for RBAC and pro 

Dagster+ uses a hierarchical model for RBAC, meaning that the most permissive roles include permissions from the roles beneath them. The following user roles are currently supported, in order from the **most** permissive to the **least** permissive:

File: docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx (hunk 6 above)
[the remark this answers is not in the document]
Comment (neverett) [reply docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx:29]: I would move this sentence to the intro, below the "In this guide..." sentence, and style it as a `<Note>`.

File: docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx [hunk 8]
```diff
@@ -14,32 +14,56 @@ In this guide, we'll cover how RBAC works in Dagster+, how to assign roles to us
...
 
-All user roles are enforced both in Dagster+ and the [GraphQL API](/concepts/webserver/graphql).
+All roles are enforced both in Dagster+ and the [GraphQL API](/concepts/webserver/graphql).
 
 ### Teams
 
 Dagster+ Pro users can create teams of users and assign default permission sets. Refer to the [Managing teams in Dagster+](/dagster-plus/account/managing-users/managing-teams) guide for more info.
 
+--- 
+## Creating and editing custom roles
```
Comment (neverett) [inline docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx:36]: I would make "Creating" and "Editing" two separate sections and go ahead and repeat the first two steps in each, to make each task a bit easier to parse.

File: docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx (hunk 3 above)
Comment (neverett) [inline docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx:47]:
```suggestion
5. Select the permissions you want to apply to the role. Note that you can base your role off of an existing role for ease of creation.
```

File: docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx [hunk 9]
```diff
@@ -14,32 +14,56 @@ In this guide, we'll cover how RBAC works in Dagster+, how to assign roles to us
...
+To edit a role:
+1. Click the edit button next to the role you want to edit
+2. Make your changes
+3. Save your changes
+4. You will see a confirmation dialog including the changes that have been made
 ---
 
-## Assigning user and team roles
+## Deleting custom roles
+Deleting custom roles is similar to editing. At the bottom of the edit dialog you'll see a button to "Delete role". Two things to note when deleting a role:
```
Comment (neverett) [inline docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx:57]: I would go ahead and list the specific steps here, to make this section a bit easier to parse for people who haven't read through the sections above, and would change the formatting of the two notes a bit. Something like the following:

```
1. Navigate to the **Organization Settings** page.
2. Click the Roles tab.
3. Click the edit button next to the role you want to edit.
4. At the bottom of the edit dialog, click **Delete role**.

Note that if the role is currently assigned to any users, you will need to reassign them to a different role before deleting the role.

<Warning>Deleting a role is a permanent action and cannot be undone.</Warning>
```

File: docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx [hunk 10]
```diff
@@ -9,37 +9,60 @@ title: Understanding role-based access control in Dagster+ | Dagster Docs
 Role-based access control (RBAC) enables you to grant specific permissions to users in your organization, ensuring that Dagster users have access to what they require in Dagster+, and no more.
 
 In this guide, we'll cover how RBAC works in Dagster+, how to assign roles to users, and the granular permissions for each user role.
+<Note> All roles are enforced both in Dagster+ and the [GraphQL API](https://docs.dagster.io/concepts/webserver/graphql) </Note>
 
 ---
 
-## Dagster+ user roles
+## Dagster+ pro user roles
```
Comment (gibsondan) [inline docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx:16]: I think we capitalize Pro usually? Is it true that only Pro plans can use RBAC?

File: docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx [hunk 11]
```diff
@@ -9,37 +9,61 @@ title: Understanding role-based access control in Dagster+ | Dagster Docs
...
+2. Click the Roles tab
+3. Click the edit button next to the role you want to edit
+4. Make your changes
+5. Save your changes
+6. You will see a confirmation dialog including the changes that have been made
 ---
 
-## Assigning user and team roles
+## Deleting custom roles
+Deleting custom roles is similar to editing. At the bottom of the edit dialog you'll see a button to "Delete role". Two things to note when deleting a role:
```
[the remark this answers is not in the document]
Comment (neverett) [reply docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx:50]: Yeah, I would change "Deleting custom roles is similar to editing. At the bottom of the edit dialog you'll see a button to "Delete role". Two things to note when deleting a role:" to something like "If you no longer need a custom role, you can delete it from the edit dialog."

File: docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx [hunk 12]
```diff
@@ -9,37 +9,60 @@ title: Understanding role-based access control in Dagster+ | Dagster Docs
...
+   - **Default roles:** Hierarchical roles, based on sensible defaults provided by Dagster.
+   - **Custom roles:** Roles you define with specific sets of permissions to match your organization's needs.
+### Teams
 
-For example, the **Admin** user role includes permissions specific to this role and all permissions in the **Editor**, **Launcher**, and **Viewer** user roles. Refer to the [User permissions reference](#user-permissions-reference) for the full list of user permissions in Dagster+.
+Dagster+ Pro users can create teams of users and assign default permission sets. Refer to the [Managing teams in Dagster+](/dagster-plus/account/managing-users/managing-teams) guide for more info.
 
-### User role enforcement
+--- 
+## Creating custom roles
```
Comment (neverett) [inline docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx:31]: Looks like many of the steps in this section and the "Editing" and "Deleting" sections need a period at the end.

File: docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx [hunk 13]
```diff
@@ -9,37 +9,60 @@ title: Understanding role-based access control in Dagster+ | Dagster Docs
...
+5. Save your changes
+6. You will see a confirmation dialog including the changes that have been made
+---
 
-All user roles are enforced both in Dagster+ and the [GraphQL API](/concepts/webserver/graphql).
+## Deleting custom roles
+Deleting custom roles is similar to editing. At the bottom of the edit dialog you'll see a button to "Delete role". Two things to note when deleting a role:
+1. Navigate to the **Organization Settings** page.
+2. Click the Roles tab.
+3. Click the edit button next to the role you want to edit.
```
[the remark this answers is not in the document]
Comment (neverett) [reply docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx:53]: agree

File: docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx (hunk 11 above)
[the remark this answers is not in the document]
Comment (steplercamp) [reply docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx:50]: I should have caught that when I first took your suggested to enumerate it fully :(

**Review by neverett — APPROVED** [review]

**Pull request merged**